autogpt_platform/backend/backend/blocks/google/gmail.py

```diff
@@ -393,37 +437,238 @@ def __init__(self):
         )
 
     async def run(
-        self, input_data: Input, *, credentials: GoogleCredentials, **kwargs
+        self,
+        input_data: Input,
+        *,
+        credentials: GoogleCredentials,
+        graph_exec_id: str,
+        user_id: str,
+        **kwargs,
     ) -> BlockOutput:
         service = GmailReadBlock._build_service(credentials, **kwargs)
         result = await asyncio.to_thread(
             self._send_email,
             service,
-            input_data.to,
-            input_data.subject,
-            input_data.body,
+            input_data,
+            graph_exec_id,
+            user_id,
         )
         yield "result", result
 
-    def _send_email(self, service, to: str, subject: str, body: str) -> dict:
-        if not to or not subject or not body:
-            raise ValueError("To, subject, and body are required for sending an email")
-        message = self._create_message(to, subject, body)
+    def _send_email(
+        self, service, input_data: Input, graph_exec_id: str, user_id: str
+    ) -> dict:
+        if (
+            not input_data.to
+            or len(input_data.to) == 0
```

Comment:
is len(input_data.to) == 0  still needed where you already have not input_data.to